scripts/run-integration-tests.sh

```diff
@@ -3,78 +3,114 @@
 ./scripts/build.sh || exit 1
 
 source "scripts/testlib.sh"
-start-monerod-regtest
-start-ganache
-start-alice-wallet
-start-bob-wallet
-start-charlie-wallet
+check-set-swap-test-data-dir
+mkdir -p "${SWAP_TEST_DATA_DIR}/"{alice,bob,charlie}
 
-CHARLIE_ETH_KEY="${SWAP_TEST_DATA_DIR}/charlie-eth.key"
+# Charlie uses deterministic ganache key #49
+CHARLIE_ETH_KEY="${SWAP_TEST_DATA_DIR}/charlie/eth.key"
 echo "87c546d6cb8ec705bea47e2ab40f42a768b1e5900686b0cecc68c0e8b74cd789" >"${CHARLIE_ETH_KEY}"
 
-# wait for wallets to start
-sleep 5
+# This is the local multiaddr created when using ./tests/alice-libp2p.key on the default libp2p port
+ALICE_MULTIADDR=/ip4/127.0.0.1/tcp/9933/p2p/12D3KooWAAxG7eTEHr2uBVw3BDMxYsxyqfKvj3qqqpRGtTfuzTuH
+ALICE_LIBP2PKEY=./tests/alice-libp2p.key
 
 start-swapd() {
 	local swapd_user="${1:?}"
 	local swapd_flags=("${@:2}")
+	local log_file="${SWAP_TEST_DATA_DIR}/${swapd_user}-swapd.log"
 	echo "Starting ${swapd_user^}'s swapd, logs in ${SWAP_TEST_DATA_DIR}/${swapd_user}-swapd.log"
-	./swapd "${swapd_flags[@]}" &>"${SWAP_TEST_DATA_DIR}/${swapd_user}-swapd.log" &
-	echo "${!}" >"${SWAP_TEST_DATA_DIR}/${swapd_user}-swapd.pid"
+	./swapd "${swapd_flags[@]}" &>"${log_file}" &
+	local swapd_pid="${!}"
+	echo "${swapd_pid}" >"${SWAP_TEST_DATA_DIR}/${swapd_user}-swapd.pid"
+	sleep 1
+	if ! kill -0 "${swapd_pid}" 2>/dev/null; then
+		echo "Failed to start ${swapd_user^}'s swapd"
+		echo "=============== Failed logs  ==============="
+		cat "${log_file}"
+		echo "============================================"
+		exit 1
+	fi
 }
 
 stop-swapd() {
 	local swapd_user="${1}"
 	stop-program "${swapd_user}-swapd"
 }
 
-start-swapd alice \
-	--dev-xmrtaker \
-	--libp2p-key=./tests/alice.key
-
-sleep 3 # Alice's swapd is a bootnode for Bob and Charlie's swapd
-
-start-swapd bob \
-	--dev-xmrmaker \
-	--bootnodes /ip4/127.0.0.1/tcp/9933/p2p/12D3KooWAAxG7eTEHr2uBVw3BDMxYsxyqfKvj3qqqpRGtTfuzTuH \
-	--wallet-file test-wallet \
-	--deploy
-
-start-swapd charlie \
-	--monero-endpoint "http://127.0.0.1:${CHARLIE_WALLET_PORT}/json_rpc" \
-	--ethereum-privkey "${CHARLIE_ETH_KEY}" \
-	--libp2p-port 9955 \
-	--rpc-port 5003 \
-	--bootnodes /ip4/127.0.0.1/tcp/9933/p2p/12D3KooWAAxG7eTEHr2uBVw3BDMxYsxyqfKvj3qqqpRGtTfuzTuH \
-	--deploy
+start-daemons() {
+	start-monerod-regtest
+	start-ganache
+
+	start-swapd alice \
+		--dev-xmrtaker \
+		"--data-dir=${SWAP_TEST_DATA_DIR}/alice" \
+		"--libp2p-key=${ALICE_LIBP2PKEY}" \
+		--deploy
+
+	#
+	# Wait up to 10 seconds for Alice's swapd instance to start and deploy the swap contract
+	#
+	CONTRACT_ADDR_FILE="${SWAP_TEST_DATA_DIR}/alice/contract-address.json"
+	for _ in {1..10}; do
+		if [[ -f "${CONTRACT_ADDR_FILE}" ]]; then
+			break
+		fi
+		sleep 1
+	done
+	if ! CONTRACT_ADDR="$(cat "${SWAP_TEST_DATA_DIR}/alice/contract-address.json")"; then
+		echo "Failed to get Alice's deployed contract address"
+		stop-daemons
+		exit 1
+	fi
+
+	start-swapd bob \
+		--dev-xmrmaker \
+		"--wallet-port=${BOB_WALLET_PORT}" \
```

Comment:
Only Bob uses this `--wallet-port` flag. As noted in `integration_tests.go` below, we needed direct access to the wallet from the integration tests to generate funds for the XMR maker. The `--wallet-port` flag is marked as hidden to discourage people from using it unnecessarily.